extensions-core/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/KubernetesPeonLifecycle.java

```diff
@@ -286,16 +290,16 @@ protected TaskLocation getTaskLocation()
         log.warn("Could not get task location from k8s for task [%s].", taskId);
         return TaskLocation.unknown();
       }
-      taskLocation = TaskLocation.create(
+      taskLocationRef.set(TaskLocation.create(
```

Comment:
I think there is still a race condition, since we are not atomically updating tasklocation reference. 

Instead, we should be updating like this:
```
taskLocationRef.compareAndSet(null, <new tasklocation>)
```

Reply:
I wanted to make last write win instead of first write win hence i think we can ignore CAS.

Reply:
even if multiple threads call taksLocationRef.set() that operation is inherently threadsafe since we are using an atomic reference. 
earlier we were using a normal variable where multiple threads are accessing this variable which leads to weird races where one thread sets the variable but its not visible to the other thread.